be/src/exec/pipeline/exchange/local_exchange.cpp

```diff
@@ -147,4 +147,111 @@ Status PassthroughExchanger::accept(const ChunkPtr& chunk, const int32_t sink_dr
 bool LocalExchanger::need_input() const {
     return !_memory_manager->is_full() && !is_all_sources_finished();
 }
+
+Status RandomPartitioner::partition_chunk(const ChunkPtr& chunk, int32_t num_partitions,
+                                          std::vector<uint32_t>& partition_row_indexes) {
+    size_t num_rows = chunk->num_rows();
+
+    _shuffle_channel_id.resize(num_rows, 0);
+    {
+        if (num_rows <= num_partitions) {
+            std::iota(_shuffle_channel_id.begin(), _shuffle_channel_id.end(), 0);
+        } else {
+            size_t i = 0;
+            for (; i < num_rows - num_partitions; i += num_partitions) {
+                std::iota(_shuffle_channel_id.begin() + i, _shuffle_channel_id.begin() + i + num_partitions, 0);
+            }
+            if (i < num_rows - 1) {
+                std::iota(_shuffle_channel_id.begin() + i, _shuffle_channel_id.end(), 0);
+            }
+        }
```

Comment:
>  We could unify if and else
when `num_rows <= num_partitions`,  `num_rows - num_partition` will be negative, it's dangerous to iterate. 
just be clear here.

> And if (i < num_rows - 1) should be if (i < num_rows)?
it makes no sense because  if `i == num_rows - 1`,  _shuffle_channel_id is just 0 directly.